From dd73d78e5e9479c979ce20b163c7a245c6ae6a43 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 24 Apr 2024 11:53:40 +0300
Subject: [PATCH] [FSW-4100] fs1: disable lbw sz check for mem copy

Disable LBW size check for Mem_Copy engine LBW transaction
per H15-255 issue.

Change-Id: Ia6885e10e98e811bba49a5948f1d1a16150b37c8
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 0c90cfd..1980baf 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -29,6 +29,8 @@
 #include <mem_copy_regs.h>
 #include <mem_copy_masks.h>
 #include <pcie_flr_regs.h>
+#include <tran_chk_lbw_regs.h>
+#include <tran_chk_lbw_masks.h>
 #include <fs1_blocks_linux_driver.h>
 #include <fs1_security.h>
 
@@ -447,6 +449,10 @@
 	regmap_writel(copy_sz, mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_MEM_COPY_BASE +
 			       mmMEM_COPY_COPY_SIZE);
 
+	/* disable LBW size check per H15-255 */
+	regmap_rmwritel(mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_MEM_COPY_MSTR_IF_LBW_TRAN_CHK_BASE +
+			mmTRAN_CHK_LBW_CHECK_EN, 0, TRAN_CHK_LBW_CHECK_EN_SIZE_CHK_M);
+
 	/* Enable copy and wait for completion */
 	regmap_writel(MEM_COPY_COPY_EN_VAL_M, mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_MEM_COPY_BASE +
 					      mmMEM_COPY_COPY_EN);
